### PR TITLE
Add navbar dropdown menus (left-side only)

### PR DESCRIPTION
**Pull request by jose-correia**
This PR adds the dropdown logic and the left-side part of the dropdown menus.

The dropdowns are loaded on the background after loading the header so that we don't impact the Lighthouse score with the files that need to be fetched.

This PR also separates the header logic into multiple files so that it's easier to navigate and improves code reuse

Fix #<gh-issue-id>

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.page/
- After: https://navbar-products-megamenu--moleculardevices--hlxsites.hlx.page/drafts/josec/

### CODE REVIEW

File: blocks/header/header.css [hunk 1]
```diff
@@ -158,7 +170,96 @@ header nav .nav-tools {
...
+  padding: 0;
+  overflow: hidden;
+  z-index: 0;
+  margin: 0 !important;
+  display: block;
+  opacity: 1;
+  visibility: visible;
+  position: absolute;
+  background: #fff url('/images/header-background-filler.jpg') no-repeat right bottom;
+  background-repeat: no-repeat;
```
Comment (mhaack) [inline blocks/header/header.css:189]: Line above already has `no-repeat` this one should not be needed.

File: blocks/header/header.css [hunk 2]
```diff
@@ -158,7 +170,96 @@ header nav .nav-tools {
...
+}
+
+.header .nav-tabs .head-submenu {
+  display: flex;
+  justify-content: space-between;
+}
+
+.header .nav-tabs .head-submenu > div {
+  background-size: 250px;
+  color: #fff;
```
Comment (mhaack) [inline blocks/header/header.css:241]: Use color variables.

File: blocks/header/header.css [hunk 3]
```diff
@@ -158,7 +170,96 @@ header nav .nav-tools {
...
+  width: 50%;
+}
+
+header .nav-tabs .menu-nav-submenu ul {
+  float: none;
+}
+
+header .nav-tabs .menu-nav-submenu-sections {
+  display: block;
+  border-right: 1px solid #FFFFFF80;
```
Comment (mhaack) [inline blocks/header/header.css:256]: Use color variables.

File: scripts/scripts.js [hunk 4]
```diff
@@ -112,6 +112,26 @@ export function addFavIcon(href, rel = 'icon') {
...
 }
 
+/**
+ * Takes a string as an input, converts it to lowercase,
+ * replaces special characters with hyphens,
+ * and removes any trailing or leading hyphens.
+ * @param {*} inputString
+ * @returns string
+ */
+export function getIdFromString(inputString) {
```
Comment (mhaack) [inline scripts/scripts.js:122]: We already have `toClassName` helper? Can't this be used here as well instead of new helper?

Comment (jose-correia) [reply scripts/scripts.js:122]: Nice! Didnt know about that one

File: blocks/header/header-events.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  if (mediaQueryList.matches) {
+    addEventListenersDesktop();
+  } else {
+    addEventListenersMobile();
+  }
+}
+
+export function handleViewportChanges(block) {
+  mediaQueryList.onchange = () => {
+    document.querySelector('main').style.visibility = '';
```
Comment (andreituicu) [inline blocks/header/header-events.js:63]: I don't fully understand this change. why do we need to update the visibility of `main`?

**Review by andreituicu — APPROVED** [review]
LGTM!

Small comments above.

File: blocks/header/header-events.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    el.replaceWith(el.cloneNode(true));
+  });
+  elementsWithEventListener = [];
+}
+
+function addEventListenersDesktop() {
+  function expandMenu(element) {
+    const expanded = element.getAttribute('aria-expanded') === 'true';
+    collapseAllSubmenus(element.closest('ul'));
+    element.setAttribute('aria-expanded', expanded ? 'false' : 'true');
```
[the remark this answers is not in the document]
Comment (mhaack) [reply blocks/header/header-events.js:19]: Not sure, not an expert here but I think setting `aria-expanded` to false is valid. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded

File: scripts/scripts.js [hunk 7]
```diff
@@ -117,7 +119,12 @@ export function addFavIcon(href, rel = 'icon') {
  */
 async function loadLazy(doc) {
   const main = doc.querySelector('main');
+
   loadHeader(doc.querySelector('header'));
+
+  // Fetch and style the megamenus in the background
+  fetchAndStyleHeaderMegamenus(doc.querySelector('header'));
```
Comment (mhaack) [inline scripts/scripts.js:126]: Thinking if we need this at all? Already `loadHeader` is called lazy, can't we merge it with that function?

File: blocks/header/header-events.js (hunk 5 above)
Comment (jose-correia) [reply blocks/header/header-events.js:63]: Good point, this logic was copied from the another Franklin block, deleting doesn't seem to have any impact at first, will remove

**Pull request merged**